addons/website_sale/controllers/main.py

```diff
@@ -1504,27 +1505,33 @@ def _get_shop_payment_values(self, order, **kwargs):
             'landing_route': '/shop/payment/validate',
         }
         values = {**portal_page_values, **payment_form_values}
-        if request.website.enabled_delivery:
-            has_storable_products = any(
-                line.product_id.type in ['consu', 'product'] for line in order.order_line
-            )
-            if not order._get_delivery_methods() and has_storable_products:
-                values['errors'].append((
-                    _('Sorry, we are unable to ship your order'),
-                    _('No shipping method is available for your current order and shipping address.'
-                    ' Please contact us for more information.')
-                ))
-
-            if has_storable_products:
-                if order.carrier_id and not order.delivery_rating_success:
-                    order._remove_delivery_line()
-                values['deliveries'] = order._get_delivery_methods().sudo()
-
-            values['delivery_has_storable'] = has_storable_products
-            values['delivery_action_id'] = request.env.ref(
-                'delivery.action_delivery_carrier_form'
-            ).id
+        values = self._get_accepted_carriers(order, values, False)
+        return values
 
+    def _get_accepted_carriers(self, order, values, express_payment, **kwargs):
+        if order.carrier_id and not order.delivery_rating_success:
```

Comment:
You removed all the `has_storable_products` part, which is needed at least for the template, even the one you add. Please add it back.